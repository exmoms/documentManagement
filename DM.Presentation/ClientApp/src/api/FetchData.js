/**
 * @file this file includes functions that fetch data from API
 */

// Generic Delete Method
export async function deleteFromApi(url) {
  let res = await fetch(url, {
    method: "DELETE",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  });
  return res;
}

// Get MetaData Models
export async function fetchData(url) {
  let object = {};
  /*
   * send the request through a CORS proxy
   * https://stackoverflow.com/questions/43262121/trying-to-use-fetch-and-pass-in-mode-no-cors/43268098#43268098
   */
  await fetch(url, {
    method: "GET",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
  })
    .then((response) => response.json())
    .then(
      (data) => (object = data),
      (error) => console.log(error)
    );
  return object;
}
