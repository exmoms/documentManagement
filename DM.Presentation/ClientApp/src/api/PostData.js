/**
 * @file This file includes functions that help to post data to a specific API
 * @version 1.0
 */

/**
 * Post new document data.
 * Upload form-data, scanned copy of doc and attachments.
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @param {*} documentData
 * @param {*} images
 * @param {*} attachments
 * @param {*} url
 * @param {*} option its values either "Add" for add new document version or "Update" to update exsiting document Version
 */
export async function postDocumentData(
  documentData,
  scannedPages,
  attachments,
  url = "",
  option
) {
  var formData = new FormData();

  if (process.env.NODE_ENV === "development") {
    console.log(documentData);
    console.log(attachments);
    console.log(scannedPages);
  }

  switch (option) {
    case "Add":
      formData.append("Document", JSON.stringify(documentData));
      break;
    case "Update":
      formData.append("Version", JSON.stringify(documentData));
      break;
    default:
  }

  attachments.forEach((attachment) => {
    formData.append("Attachments", attachment);
  });
  scannedPages.forEach((scannedPage) => {
    formData.append("Scans", scannedPage);
  });

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    headers: {
      Accept: "application/json",
      // "Content-Type": "application/json",
    },
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Attachment Data.
 * Upload attachment json data + attachment file  .
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} payload
 * @param {*} attachmentFile
 * @param {*} url
 */
export async function postAttachmentData(payload, attachmentFile, url = "") {
  var formData = new FormData();

  formData.append("Attachment", JSON.stringify(payload));
  formData.append("FileAttachment", attachmentFile);

  let response = {};
  await fetch(url, {
    // content-type header should not be specified!
    method: "POST",
    body: formData,
  }).then(
    (res) => {
      response = res;
      // response.json();
    },
    (error) => console.log("[ERROR]:", error)
  );
  return response;
}

/**
 * Post Request To API
 * Send post request to api with  payload  and return a response.
 * @author Ali Daghman <ali.daghman@lit-co.net>
 * @param {*} url The api url
 * @param {*} payload js object which will be sent as request body.
 */
export async function postDataToAPI(url, payload = null, method = "POST") {

  const response = await fetch(url, {
    method: method,
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(payload),
  });
  return response;
}
