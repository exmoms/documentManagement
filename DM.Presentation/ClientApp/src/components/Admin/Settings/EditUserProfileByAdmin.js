import { FormHelperText } from "@material-ui/core";
import Container from "@material-ui/core/Container";
import CssBaseline from "@material-ui/core/CssBaseline";
import FormControl from "@material-ui/core/FormControl";
import Grid from "@material-ui/core/Grid";
import IconButton from "@material-ui/core/IconButton";
import InputAdornment from "@material-ui/core/InputAdornment";
import InputLabel from "@material-ui/core/InputLabel";
import OutlinedInput from "@material-ui/core/OutlinedInput";
import { makeStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Visibility from "@material-ui/icons/Visibility";
import VisibilityOff from "@material-ui/icons/VisibilityOff";
import clsx from "clsx";
import React, { useEffect, useState } from "react";
// import { postData } from "../../api/PostData";
import { useTranslation } from "react-i18next";

const useStyles = makeStyles((theme) => ({
  paper: {
    marginTop: theme.spacing(8),
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
  form: {
    width: "100%", // Fix IE 11 issue.
    marginTop: theme.spacing(3),
  },
  submit: {
    margin: theme.spacing(3, 0, 2),
  },
  margin: {
    margin: theme.spacing(1),
  },
  textField: {
    width: "100%",
  },
}));

export default function EditUserProfileByAdmin(props) {
  const { t } = useTranslation();
  const classes = useStyles();

  const [email, setemail] = useState("");
  const [tel, settel] = useState("");
  const [values, setValues] = useState({
    password: "",
    showPassword: false,
  });

  useEffect(() => {
    if (props.profileData !== null) {
      setemail(props.profileData.email);
      settel(
        props.profileData.phoneNumber === null
          ? ""
          : props.profileData.phoneNumber
      );
    }
  }, [props]);

  const onChange = (event) => {
    if (event.target.name === "edit-profile-by-admin-email") {
      setemail(event.target.value);
    }
    if (event.target.name === "edit-profile-by-admin-tel") {
      settel(event.target.value);
    }
  };

  const handleChange = (prop) => (event) => {
    setValues({ ...values, [prop]: event.target.value });
  };

  const handleClickShowPassword = () => {
    setValues({ ...values, showPassword: !values.showPassword });
  };

  const handleMouseDownPassword = (event) => {
    event.preventDefault();
  };

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Grid container spacing={1}>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              required
              fullWidth
              value={email}
              onChange={onChange}
              name="edit-profile-by-admin-email"
              label={t("email")}
              type="email"
              id="edit-profile-by-admin-email"
              autoComplete="email"
            />
          </Grid>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              fullWidth
              value={tel}
              onChange={onChange}
              name="edit-profile-by-admin-tel"
              label={t("phone_number")}
              type="tel"
              id="edit-profile-by-admin-tel"
              autoComplete="tel"
            />
          </Grid>
          <FormControl
            className={clsx(classes.margin, classes.textField)}
            variant="outlined"
          >
            <InputLabel
              error={!!props.errors.newPasswordErrors}
              htmlFor="edit-profile-by-admin-outlined-adornment-password"
            >
              {t("new_password")}
            </InputLabel>
            <OutlinedInput
              error={!!props.errors.newPasswordErrors}
              id="edit-profile-by-admin-outlined-adornment-password"
              type={values.showPassword ? "text" : "password"}
              value={values.password}
              onChange={handleChange("password")}
              endAdornment={
                <InputAdornment position="end">
                  <IconButton
                  id="iconButton"
                    aria-label="toggle password visibility"
                    onClick={handleClickShowPassword}
                    onMouseDown={handleMouseDownPassword}
                    edge="end"
                  >
                    {values.showPassword ? <Visibility /> : <VisibilityOff />}
                  </IconButton>
                </InputAdornment>
              }
              labelWidth={120}
            />
            <FormHelperText
              error={!!props.errors.newPasswordErrors}
              id="edit-profile-by-admin-outlined-adornment-password-helper-text"
            >
              {props.errors.newPasswordErrors}
            </FormHelperText>
          </FormControl>
          <Grid item xs={12}>
            <TextField
              variant="outlined"
              fullWidth
              name="edit-profile-by-admin-confirmPassword"
              label={t("confirmNewPassword_label")}
              type="password"
              error={!!props.errors.confirmErrors}
              helperText={props.errors.confirmErrors}
              id="edit-profile-by-admin-confirmPassword"
              autoComplete="current-password"
            />
          </Grid>
        </Grid>
      </div>
    </Container>
  );
}
