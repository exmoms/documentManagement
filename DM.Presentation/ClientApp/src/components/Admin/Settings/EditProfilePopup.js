/**
 * @file Contains BrowseAggregatedDocsPopUp component implementaion
 * @author Youssef Shaaban <y.shaaban@lit-co.net>
 * @version 1.0
 */

import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import Paper from "@material-ui/core/Paper";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableRow from "@material-ui/core/TableRow";
import Typography from "@material-ui/core/Typography";
import React, { Component } from "react";
import { withTranslation } from "react-i18next";
import { fetchData } from "../../../api/FetchData";
import { postDataToAPI } from "../../../api/PostData";
import { Notification } from "../Notifications";
import EditUserProfileByAdmin from "./EditUserProfileByAdmin";

const styles = (theme) => ({
  paper: {
    maxWidth: "auto",
  },
  margin: {
    margin: theme.spacing(2),
  },
});

class EditProfilePopup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: this.props.user,
      userProfile: null,
      errors: {
        confirmErrors: "",
        newPasswordErrors: "",
        submissionErrors: "",
      },
      showSuccess: false,
    };
    this.updateParentState = this.props.action;
  }
  t = this.props["t"];

  componentDidUpdate(prevProps) {
    if (prevProps.user.userId !== this.props.user.userId) {
      fetchData(`/api/user/${this.props.user.userId}`).then((profileData) => {
        this.setState({ userProfile: profileData });
      });
    }
  }

  toggleShow = (show) => {
    this.updateParentState(show);
  };

  handelSubmit = (event) => {
    event.preventDefault();
    let email = document.getElementById("edit-profile-by-admin-email").value;
    let tel = document.getElementById("edit-profile-by-admin-tel").value;
    let newPassword = document.getElementById(
      "edit-profile-by-admin-outlined-adornment-password"
    ).value;
    let confirmPassword = document.getElementById(
      "edit-profile-by-admin-confirmPassword"
    ).value;
    let validPassword = false;
    if (newPassword === "") {
      validPassword = newPassword === confirmPassword;
    } else {
      validPassword =
        newPassword === confirmPassword &&
        /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\w]).{8,}$/i.test(newPassword);
    }

    if (validPassword) {
      const formData = {
        userId: this.props.user.userId,
        email: email,
        phoneNumber: tel,
        NewPassword: newPassword,
        ConfirmPassword: confirmPassword,
      };
      postDataToAPI("/api/user/EditUserProfileByAdmin", formData)
        .then(
          (res) => {
            if (res.ok) {
              this.setState({
                errors: {
                  confirmErrors: "",
                  newPasswordErrors: "",
                  submissionErrors: "",
                },
                showSuccess: true,
              });
            }
            return res.json();
          },
          (error) => console.log(error)
        )
        .then(
          (data) => {
            this.setState({
              errors: {
                confirmErrors: "",
                newPasswordErrors: "",
                submissionErrors: data.error,
              },
              showSuccess: true,
            });
          },
          (error) => console.log(error)
        );

      this.toggleShow(false);
    } else {
      this.setState({
        errors: {
          confirmErrors:
            newPassword === confirmPassword
              ? ""
              : this.t("password_mismatchingWithComfirm"),
          newPasswordErrors: /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[^\w]).{8,}$/i.test(
            newPassword
          )
            ? ""
            : this.t("password_formateError"),
        },
      });
    }
  };

  render() {
    const { classes } = this.props;
    return (
      <div>
        <Dialog
          position="centerCenter"
          open={this.props.show}
          onClose={() => this.toggleShow(false)}
          aria-labelledby="form-dialog-title"
          className={classes.paper}
        >
          <DialogTitle id="form-dialog-title">
            <Typography variant="h6" gutterBottom align="left">
              {this.t("edit_user_profile")} : {this.props.user.userName}
            </Typography>
          </DialogTitle>
          <form
            className={classes.form}
            id="edit-profile-form"
            onSubmit={this.handelSubmit}
          >
            <DialogContent>
              <TableContainer component={Paper}>
                <Table className={classes.table} aria-label="simple table">
                  <TableBody>
                    <TableRow key="search-form">
                      <TableCell>
                        <EditUserProfileByAdmin
                          profileData={this.state.userProfile}
                          errors={this.state.errors}
                        />
                      </TableCell>
                    </TableRow>
                  </TableBody>
                </Table>
              </TableContainer>
            </DialogContent>
            <DialogActions>
              <Button
                id="submit-btn"
                type="submit"
                // variant="contained"
                color="primary"
                className={classes.submit}
              >
                {this.t("submit_changes")}
              </Button>
              <Button
                id="CancelButton"
                onClick={() => this.toggleShow(false)}
                color="primary"
              >
                {this.t("button_cancel")}
              </Button>
            </DialogActions>
          </form>
        </Dialog>
        <Notification
          id="notification"
          open={this.state.showSuccess}
          setOpen={(show) => this.setState({ showSuccess: show })}
          error={!!this.state.errors.submissionErrors}
          errorMessage={this.state.errors.submissionErrors}
        />
      </div>
    );
  }
}

export default withStyles(styles)(withTranslation()(EditProfilePopup));
