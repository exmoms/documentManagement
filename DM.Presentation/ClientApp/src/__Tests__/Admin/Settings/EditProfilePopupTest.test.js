import React from "react";
import ReactDom from "react-dom";
import EditProfilePopup from "../../../components/Admin/Settings/EditProfilePopup";
import { cleanup } from "@testing-library/react";
import { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";
import TextField from "@material-ui/core/TextField";
import { act } from "react-dom/test-utils";
import Button from "@material-ui/core/Button";
import { BrowserRouter as Router } from "react-router-dom";
import * as FetchMock from "../../../api/FetchData";
import * as PostDataMock from "../../../api/PostData";
import Typography from "@material-ui/core/Typography";
import Dialog from "@material-ui/core/Dialog";
import EditUserProfileByAdmin from "../../../components/Admin/Settings/EditUserProfileByAdmin";
afterEach(cleanup);
describe("EditProfilePopup", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  test("renders without crashing", () => {
    const div = document.createElement("div");
    let user={userId:1 , userName:"userName A"};
    let show= true;
    const action = jest.fn();
    ReactDom.render(
        <EditProfilePopup user={user} show={show} action={action}/>,
      div
    );
  });

  test("matches snapshot", () => {
    let user={userId:1 , userName:"userName A"};
    let show= true;
    const action = jest.fn();
    const wrapper = shallow(
        <EditProfilePopup user={user} show={show} action={action} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
   
  test("intialization ProfilePopUp", () => {
    let user={userId:1 , userName:"userName A"};
    let show= true;
    const action = jest.fn();
  let  wrapper = mount(<EditProfilePopup user={user} show={show} action={action} />) ;
  expect(wrapper.find(Dialog).at(0).props().open).toBeTruthy();
  expect(wrapper.find(Typography).at(0).text()).toBe("edit_user_profile : userName A");
  expect(wrapper.find(EditUserProfileByAdmin).at(0).props().profileData).toBe(null);
  expect(wrapper.find(EditUserProfileByAdmin).at(0).props().errors).toStrictEqual({ confirmErrors: "", newPasswordErrors: "",  submissionErrors: ""});
  expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
  expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual("");
});

test("componentDidUpdate", () => {
    let user={userId:1 , userName:"userName A"};
    let user2={userId:2 , userName:"userName B"};
    let show= true;
    const action = jest.fn();
    const mockSuccessResponse = {email:"test@hotmail.com" , phoneNumber:9888000000};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const fetchMock =  jest.spyOn(FetchMock, "fetchData").mockImplementation(() => mockJsonPromise);
    let wrapper;
    return mockJsonPromise.then(() => {
      wrapper = mount(<EditProfilePopup user={user} show={show} action={action} />) ;
      wrapper.setProps({user:user2});
    }).then(() => {
        wrapper.render();
        wrapper.update();
        expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(fetchMock.mock.calls[0][0]).toBe("/api/user/2");
      expect(wrapper.find(Dialog).at(0).props().open).toBeTruthy();
      expect(wrapper.find(Typography).at(0).text()).toBe("edit_user_profile : userName B");
      expect(wrapper.find(EditUserProfileByAdmin).at(0).props().profileData).toStrictEqual({email:"test@hotmail.com" , phoneNumber:9888000000});
      expect(wrapper.find(EditUserProfileByAdmin).at(0).props().errors).toStrictEqual({ confirmErrors: "", newPasswordErrors: "",submissionErrors: ""});
      expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual("");
    });
});

test("toggleShow", () => {
    let user={userId:1 , userName:"userName A"};
    let show= true;
    const action = jest.fn();
    let wrapper;
    wrapper = mount(<EditProfilePopup user={user} show={show} action={action} />) ;
    act(() => {
        wrapper
          .find(Dialog)
          .at(0)
          .props()
          .onClose();
      }); 
      wrapper.render();
      wrapper.update();
      expect(action).toHaveBeenCalledTimes(1);
      expect(action.mock.calls[0][0]).toBeFalsy();
});

test("handelSubmit when res is ok", () => {
    let user={userId:1 , userName:"userName A"};
    let user2={userId:2 , userName:"userName B"};
    let show= true;
    const action = jest.fn();
    const preventDefaultMock = jest.fn();
    const mockSuccessResponse = {email:"test@hotmail.com" , phoneNumber:9888000000};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse);
    const fetchMock =  jest.spyOn(FetchMock, "fetchData").mockImplementation(() => mockJsonPromise);
    const mockSuccessResponsePostDataOk = Promise.resolve({ok:true, json:()=>{return  Promise.resolve( {error:["error"]})}});
    const mockSuccessResponsePostDataNotOk = Promise.resolve({ok:false, json:()=>{return  Promise.resolve( {error:["error"]})}});
    const PostMock=  jest.spyOn(PostDataMock, "postDataToAPI").mockReturnValueOnce(mockSuccessResponsePostDataOk).mockReturnValueOnce(mockSuccessResponsePostDataNotOk);
    const formData = {
        userId: 2,
        email: "test.value@hotmail.com",
        phoneNumber: "9858444444",
        NewPassword: "",
        ConfirmPassword: "",
      };
      const formDataNotOk = {
        userId: 2,
        email: "test@hotmail.com",
        phoneNumber: "9888000000",
        NewPassword: "",
        ConfirmPassword: "",
      };
    let wrapper;
    return mockJsonPromise.then(() => {
      wrapper = mount(<EditProfilePopup user={user} show={show} action={action} /> ,{ attachTo: document.body }) ;
      wrapper.setProps({user:user2});
    }).then(() => {
        wrapper.render();
        wrapper.update();
        expect(fetchMock).toHaveBeenCalledTimes(1);
      expect(fetchMock.mock.calls[0][0]).toBe("/api/user/2");
      return mockSuccessResponsePostDataOk.then(() => {
        document.getElementById("edit-profile-by-admin-email").value="test.value@hotmail.com";
        document.getElementById("edit-profile-by-admin-tel").value="9858444444";
        document.getElementById("edit-profile-by-admin-outlined-adornment-password").value="";
        document.getElementById("edit-profile-by-admin-confirmPassword").value="";
        expect(wrapper.find("#notification").at(0).props().open).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual("");
        act(() => {
            wrapper
              .find("#edit-profile-form")
              .at(0)
              .props()
              .onSubmit({ preventDefault: preventDefaultMock });
          }); 
    }).then(() => {
        wrapper.render();
      wrapper.update();
      expect(preventDefaultMock).toHaveBeenCalledTimes(1);
      expect(PostMock).toHaveBeenCalledTimes(1);
      expect(PostMock.mock.calls[0][0]).toBe("/api/user/EditUserProfileByAdmin");
      expect(PostMock.mock.calls[0][1]).toStrictEqual(formData);
      expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
    }).then(() => {
        wrapper.render();
      wrapper.update();
      expect(action).toHaveBeenCalledTimes(1);
      expect(action.mock.calls[0][0]).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual("");
      //when the res is not ok 
      return mockSuccessResponsePostDataNotOk.then(() => {
        // document.getElementById("edit-profile-by-admin-email").value="test.value@hotmail.com";
        // document.getElementById("edit-profile-by-admin-tel").value="9858444444";
        // document.getElementById("edit-profile-by-admin-outlined-adornment-password").value="";
        // document.getElementById("edit-profile-by-admin-confirmPassword").value="";
        expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
      expect(wrapper.find("#notification").at(0).props().error).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual("");
        act(() => {
            wrapper
              .find("#edit-profile-form")
              .at(0)
              .props()
              .onSubmit({ preventDefault: preventDefaultMock });
          }); 
    }).then(() => {
        wrapper.render();
      wrapper.update();
      expect(preventDefaultMock).toHaveBeenCalledTimes(2);
      expect(PostMock).toHaveBeenCalledTimes(2);
      expect(PostMock.mock.calls[1][0]).toBe("/api/user/EditUserProfileByAdmin");
      expect(PostMock.mock.calls[1][1]).toStrictEqual(formDataNotOk);
      expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
    }).then(() => {
        wrapper.render();
      wrapper.update();
      expect(action).toHaveBeenCalledTimes(2);
      expect(action.mock.calls[0][0]).toBeFalsy();
      expect(action.mock.calls[1][0]).toBeFalsy();
      expect(wrapper.find("#notification").at(0).props().open).toBeTruthy();
      expect(wrapper.find("#notification").at(0).props().error).toBeTruthy();
      expect(wrapper.find("#notification").at(0).props().errorMessage).toStrictEqual(["error"]);
      //when the password is unvalid 
      //test case 1
      document.getElementById("edit-profile-by-admin-email").value="test.value@hotmail.com"; 
      document.getElementById("edit-profile-by-admin-tel").value="9858444444"; 
      document.getElementById("edit-profile-by-admin-outlined-adornment-password").value="..///.";
       document.getElementById("edit-profile-by-admin-confirmPassword").value="password2";
        wrapper .find("#edit-profile-form") .at(0) .props() .onSubmit({ preventDefault: preventDefaultMock }); 
        wrapper.render();
        wrapper.update();
    expect(preventDefaultMock).toHaveBeenCalledTimes(3);
     expect(PostMock).toHaveBeenCalledTimes(2);
     expect(wrapper.find(EditUserProfileByAdmin).at(0).props().errors).toStrictEqual({"confirmErrors": "password_mismatchingWithComfirm",
     "newPasswordErrors": "password_formateError"});
  
    });      
     }); 
    }); 
    });
});