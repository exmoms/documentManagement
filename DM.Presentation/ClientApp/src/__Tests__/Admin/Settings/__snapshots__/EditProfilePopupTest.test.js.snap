// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfilePopup matches snapshot 1`] = `
<withI18nextTranslation(EditProfilePopup)
  action={[MockFunction]}
  classes={
    Object {
      "margin": "withI18nextTranslation(EditProfilePopup)-margin-2",
      "paper": "withI18nextTranslation(EditProfilePopup)-paper-1",
    }
  }
  show={true}
  user={
    Object {
      "userId": 1,
      "userName": "userName A",
    }
  }
/>
`;
